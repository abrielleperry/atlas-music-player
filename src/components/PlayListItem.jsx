import React from "react";

export default function PlayListItem() {
  return (
    <div>
      <div className="font-light tracking-wide shadow-md text-xl  subpixel-antialiased text-white p-3 mb-2 rounded-t-lg bg-green dark:bg-golden-yellow  ">
        Playlist
      </div>

      <div className="flex flex-col gap-y-4">
        {[
          { title: "Painted in Blue", artist: "Soul Canvas", length: "5:55" },
          { title: "Title Drift", artist: "Echos of Seas", length: "8:02" },
          { title: "Fading Shadows", artist: "The Emberlight", length: "3:01" },
          { title: "Cosmic Drift", artist: "Solar Flare", length: "5:01" },
          {
            title: "Urban Serenade",
            artist: "Midnight Groove",
            length: "4:54",
          },
          {
            title: "Whispers in the Wind",
            artist: "Rust & Ruin",
            length: "6:13",
          },
          { title: "Electric Fever", artist: "Neon Jungle", length: "8:41" },
          {
            title: "Edge of the Abyss",
            artist: "Steel Horizon",
            length: "2:27",
          },
          { title: "Golden Haze", artist: "Velvet Waves", length: "3:15" },
          {
            title: "Shatter the Silence",
            artist: "Thunderclap Echo",
            length: "8:22",
          },
        ].map((song, index) => (
          <div
            key={index}
            className="flex items-center bg-neutral-50 dark:bg-white shadow-md hover:scale-105 p-1 rounded-lg transition duration-200 px-4"
          >
            <div className="flex-1">
              <p className="font-thin subpixel-antialiased text-base text-black">
                {song.title}
              </p>
              <p className="text-sm font-normal subpixel-antialiased  text-natural-stone  dark:text-neutral-700">
                {song.artist}
              </p>
            </div>
            <div className="text-right  text-sm font-normal subpixel-antialiased  text-natural-stone dark:text-neutral-700">
              {song.length}
            </div>
          </div>
        ))}
      </div>
    </div>
  );
}
