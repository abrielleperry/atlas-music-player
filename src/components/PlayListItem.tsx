import React from "react";

type PlayListItemProps = {
  song: {
    id: string;
    title: string;
    artist: string;
    duration: number;
  };
  isSelected: boolean;
  onClick: () => void;
};

const PlayListItem: React.FC<PlayListItemProps> = ({
  song,
  isSelected,
  onClick,
}) => {
  return (
    <div
      onClick={onClick}
      className={`playlist-item ${
        isSelected
          ? "flex items-center bg-neutral-50 dark:bg-white shadow-md hover:scale-105 p-1 rounded-lg transition duration-200 px-4"
          : "flex items-center bg-neutral-50 dark:bg-white shadow-md hover:scale-105 p-1 rounded-lg transition duration-200 px-4"
      }
        cursor-pointer`}
    >
      <div className="flex-1">
        <p className="font-thin subpixel-antialiased text-base text-black">
          {song.title}
        </p>
        <p className="text-sm font-normal subpixel-antialiased  text-natural-stone  dark:text-neutral-700">
          {song.artist}
        </p>
      </div>
      <div className="text-right  text-sm font-normal subpixel-antialiased  text-natural-stone dark:text-neutral-700">
        {Math.floor(song.duration / 60)}:{song.duration % 60}
      </div>
    </div>
  );
};

export default PlayListItem;
