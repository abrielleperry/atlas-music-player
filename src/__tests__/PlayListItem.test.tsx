import { render } from "@testing-library/react";
import "@testing-library/jest-dom/vitest";
import { expect, test, vi } from "vitest";
import PlayListItem from "../components/PlayListItem";

test("renders PlayListItem component correctly", () => {
  const mockOnClick = vi.fn();
  const song = {
    id: "soammx6oibpan244my4toqke",
    title: "Tidal Drift",
    artist: "Echos of the Sea",
    duration: 240,
  };

  const { asFragment } = render(
    <PlayListItem song={song} isSelected={false} onClick={mockOnClick} />
  );
  expect(asFragment()).toMatchSnapshot();
});

test("renders PlayListItem component with selected state", () => {
  const mockOnClick = vi.fn();
  const song = {
    id: "a6pkp78whsyqdvpb5dxn64ss",
    title: "Fading Shadows",
    artist: "The Emberlight",
    duration: 240,
  };

  const { asFragment } = render(
    <PlayListItem song={song} isSelected={true} onClick={mockOnClick} />
  );
  expect(asFragment()).toMatchSnapshot();
});

test("renders PlayListItem component with different song", () => {
  const mockOnClick = vi.fn();
  const song = {
    id: "hdees11mmk6g078ewijlly1r",
    title: "Cosmic Drift",
    artist: "Solar Flare",
    duration: 300,
  };

  const { asFragment } = render(
    <PlayListItem song={song} isSelected={false} onClick={mockOnClick} />
  );
  expect(asFragment()).toMatchSnapshot();
});
