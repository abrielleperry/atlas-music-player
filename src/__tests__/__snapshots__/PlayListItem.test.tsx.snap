// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders PlayListItem component correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-y-4 py-2"
  >
    <div
      class="flex items-center bg-neutral-50 dark:bg-white shadow-md hover:scale-105 p-2 rounded-lg transition duration-200 px-4 "
    >
      <div
        class="flex-1"
      >
        <p
          class="font-thin subpixel-antialiased text-base text-black"
        >
          Tidal Drift
        </p>
        <p
          class="text-sm font-normal subpixel-antialiased  text-natural-stone  dark:text-neutral-700"
        >
          Echos of the Sea
        </p>
      </div>
      <div
        class="text-right  text-sm font-normal subpixel-antialiased  text-natural-stone dark:text-neutral-700"
      >
        4:00 
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders PlayListItem component with different song 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-y-4 py-2"
  >
    <div
      class="flex items-center bg-neutral-50 dark:bg-white shadow-md hover:scale-105 p-2 rounded-lg transition duration-200 px-4 "
    >
      <div
        class="flex-1"
      >
        <p
          class="font-thin subpixel-antialiased text-base text-black"
        >
          Cosmic Drift
        </p>
        <p
          class="text-sm font-normal subpixel-antialiased  text-natural-stone  dark:text-neutral-700"
        >
          Solar Flare
        </p>
      </div>
      <div
        class="text-right  text-sm font-normal subpixel-antialiased  text-natural-stone dark:text-neutral-700"
      >
        5:00 
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders PlayListItem component with selected state 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-y-4 py-2"
  >
    <div
      class="flex items-center bg-neutral-50 dark:bg-white shadow-md hover:scale-105 p-2 rounded-lg transition duration-200 px-4 border-2 border-olive-green"
    >
      <div
        class="flex-1"
      >
        <p
          class="font-thin subpixel-antialiased text-base text-black"
        >
          Fading Shadows
        </p>
        <p
          class="text-sm font-normal subpixel-antialiased  text-natural-stone  dark:text-neutral-700"
        >
          The Emberlight
        </p>
      </div>
      <div
        class="text-right  text-sm font-normal subpixel-antialiased  text-natural-stone dark:text-neutral-700"
      >
        4:00 
      </div>
    </div>
  </div>
</DocumentFragment>
`;
